tests/testthat/test-helper_functions.R

```diff
@@ -0,0 +1,14 @@
+test_that("check_and_prepare_by_group() returns expected value", {
+  expect_identical(check_and_prepare_by_group(by_group = NULL), NULL)
+  expect_identical(check_and_prepare_by_group(by_group = "NULL"), NULL)
+  expect_identical(check_and_prepare_by_group(by_group = "foo"), "foo")
+  expect_identical(check_and_prepare_by_group(by_group = "foo, bar"), c("foo", "bar"))
```

Comment:
off-topic comment: I find it odd that `strsplit()` also strips out spaces (and other whitespace?) even without explicitly asking it to do so... and apparently no way to tell it not to do that